mne/decoding/tests/test_receptive_field.py

```diff
@@ -469,4 +470,62 @@ def test_receptive_field_nd():
         assert_true(score > 0.6, msg=score)
 
 
+@requires_version('sklearn', '0.17')
+def test_inverse_coef():
+    """Test inverse coefficients computation."""
+    from sklearn.linear_model import Ridge
+
+    rng = np.random.RandomState(0)
+    tmin, tmax = 0., 10.
+    n_feats, n_targets, n_samples = 64, 2, 10000
+    n_delays = int((tmax - tmin) + 1)
+
+    def make_data(n_feats, n_targets, n_samples, tmin, tmax):
+        X = rng.randn(n_samples, n_feats)
+        w = rng.randn(int((tmax - tmin) + 1) * n_feats, n_targets)
+        # Delay inputs
+        X_del = np.concatenate(
+            _delay_time_series(X, tmin, tmax, 1.).transpose(2, 0, 1), axis=1)
+        y = np.dot(X_del, w)
+        return X, y
+
+    # Check coefficient dims, for all estimator types
+    X, y = make_data(n_feats, n_targets, n_samples, tmin, tmax)
+    tdr = TimeDelayingRidge(tmin, tmax, 1., 0.1, 'laplacian')
+    for estimator in (0., 0.01, Ridge(alpha=0.), tdr):
+        rf = ReceptiveField(tmin, tmax, 1., estimator=estimator,
+                            patterns=True)
+        rf.fit(X, y)
+        inv_rf = ReceptiveField(tmin, tmax, 1., estimator=estimator,
+                                patterns=True)
+        inv_rf.fit(y, X)
+
+        assert_array_equal(rf.coef_.shape, rf.patterns_.shape,
+                           (n_targets, n_feats, n_delays))
+        assert_array_equal(inv_rf.coef_.shape, inv_rf.patterns_.shape,
+                           (n_feats, n_targets, n_delays))
+
+        # we should have np.dot(patterns.T,coef) ~ np.eye(n)
+        c0 = rf.coef_.reshape(n_targets, n_feats * n_delays)
+        c1 = rf.patterns_.reshape(n_targets, n_feats * n_delays)
+        assert_allclose(np.dot(c0, c1.T), np.eye(c0.shape[0]), atol=0.1)
+
+        # TODO this fails.
+        # Shouldn't we have rf.coef_ ~ inv_rf.patterns_ and vice versa ?
+        # assert_allclose(inv_rf.coef_, np.transpose(rf.patterns_, (1, 0, 2)),
+        #                 rtol=0.1)
+        # assert_allclose(rf.coef_, np.transpose(inv_rf.patterns_, (1, 0, 2)),
+        #                 atol=0.1)
```

Comment:
This might only be true if regularization is 0., can you check?

Reply:
yeah my sense is that the regularization will do some strange things. Plus we're including extra information in there when we do the inverse transform. If we just do encoding model ridge, we're getting weights by normalizing w/ the stimulus covariance matrix, if we do an inverted decoding model, we're getting weights normalized w/ the channels covariance matrix, then normalized again by the stimulus covariance matrix, no?

Reply:
@larsoner Yeah I checked and this also fails without regularization. Like @choldgraf I tend to think that they're something going on with the regularisation, although I can't put my finger on what exactly.